### PR TITLE
[Enhancement] container run as non root

### PR DESCRIPTION
**Pull request by yandongxiao**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docker/dockerfiles/be/be-ubi.Dockerfile [hunk 1]
```diff
@@ -43,3 +43,12 @@ COPY docker/dockerfiles/be/*.sh $STARROCKS_ROOT/
 
 # Create directory for BE storage, create cn symbolic link to be
 RUN touch /.dockerenv && mkdir -p $STARROCKS_ROOT/be/storage && ln -sfT be $STARROCKS_ROOT/cn
+
+# Run as starrocks user
+ARG USER=starrocks
+ARG GROUP=starrocks
+RUN groupadd --gid 1000 $GROUP && useradd --home-dir /nonexistent --uid 1000 --gid 1000 \
```
Comment (kevincai) [inline docker/dockerfiles/be/be-ubi.Dockerfile:50]:
```suggestion
RUN groupadd --gid 1000 $GROUP && useradd --no-create-home --uid 1000 --gid 1000 \
```

File: docker/dockerfiles/fe/fe-ubuntu.Dockerfile [hunk 2]
```diff
@@ -43,3 +43,12 @@ COPY docker/dockerfiles/fe/*.sh $STARROCKS_ROOT/
...
 RUN touch /.dockerenv && mkdir -p /opt/starrocks/fe/meta
+
+# Run as starrocks user
+ARG USER=starrocks
+ARG GROUP=starrocks
+RUN groupadd --gid 1000 $GROUP && useradd --home-dir /nonexistent --uid 1000 --gid 1000 \
+             --shell /usr/sbin/nologin $USER  \
+    && chown -R $USER:$GROUP /opt/starrocks
+USER $USER
+ENV USER $USER
```
Comment (kevincai) [inline docker/dockerfiles/fe/fe-ubuntu.Dockerfile:54]: what does this `ENV USER $USER` for?

Comment (yandongxiao) [reply docker/dockerfiles/fe/fe-ubuntu.Dockerfile:54]: no special purpose, just set it like linux system does

Comment (kevincai) [reply docker/dockerfiles/fe/fe-ubuntu.Dockerfile:54]: ok

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docker/dockerfiles/be/be-ubuntu.Dockerfile [hunk 3]
```diff
@@ -44,3 +44,12 @@ COPY docker/dockerfiles/be/*.sh $STARROCKS_ROOT/
...
 RUN touch /.dockerenv && mkdir -p $STARROCKS_ROOT/be/storage && ln -sfT be $STARROCKS_ROOT/cn
+
+# Run as starrocks user
+ARG USER=starrocks
+ARG GROUP=starrocks
+RUN groupadd --gid 1000 $GROUP && useradd --no-create-home --uid 1000 --gid 1000 \
+             --shell /usr/sbin/nologin $USER  \
+    && chown -R $USER:$GROUP /opt/starrocks
+USER $USER
+ENV USER $USER
```
Comment (kevincai) [inline docker/dockerfiles/be/be-ubuntu.Dockerfile:55]:
```suggestion
ENV USER=$USER
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by kevincai — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**